nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -567,7 +567,8 @@ protected void setACP(Document doc, ACP newAcp, boolean overwrite, Map<String, S
         if (options == null) {
             options = new HashMap<>();
         }
-        options.put(CoreEventConstants.OLD_ACP, docModel.getACP().clone());
+        ACP oldAcp = docModel.getACP();
+        options.put(CoreEventConstants.OLD_ACP, oldAcp == null ? null : oldAcp.clone());
```

Comment:
I wonder if you even need to set the `CoreEventConstants.OLD_ACP` option in case `oldAcp` is null?

Reply:
Only usage is in `PermissionListener` which does nothing if `oldAcp` is null. Except the question should we do something in case of null in `PermissionListener`. I would go to not put something null in options.